src/shrink.rs

```diff
@@ -25,3 +27,65 @@ pub fn shrink(m: &mut Module) {
     }
     passes::gc::run(m);
 }
+
+pub fn optimize(m: &mut Module, keep_name_section: bool, level: &str) -> anyhow::Result<()> {
```

Comment:
`keep_name_section` is not needed. It should be always `false` when asked. The goal of using wasm-opt is to reduce the wasm size, so it's safe to exclude the name section.

Also rename the function to `shrink_with_wasm_opt`?